ash/src/vk/definitions.rs

```diff
@@ -51890,9 +51890,22 @@ unsafe impl<'a> TaggedStructure for GetLatencyMarkerInfoNV<'a> {
     const STRUCTURE_TYPE: StructureType = StructureType::GET_LATENCY_MARKER_INFO_NV;
 }
 impl<'a> GetLatencyMarkerInfoNV<'a> {
+    #[doc = r" This only sets the slice pointer. Users must manually pass the length of"]
+    #[doc = r" `timings` to [`crate::extensions::nv::LowLatency2::get_latency_timings()`]."]
+    #[doc = r""]
+    #[doc = r" [`Self::p_timings`] will be set to to [`std::ptr::null_mut()`] if the"]
+    #[doc = r" slice is empty, which is necessary to query the length of the array via"]
+    #[doc = r" [`crate::extensions::nv::LowLatency2::get_latency_timings_len()`]."]
+    #[doc = r""]
+    #[doc = r" See <https://registry.khronos.org/vulkan/specs/1.3-extensions/man/html/vkGetLatencyTimingsNV.html#_description>"]
+    #[doc = r" for more details."]
     #[inline]
-    pub fn timings(mut self, timings: &'a mut LatencyTimingsFrameReportNV<'a>) -> Self {
-        self.p_timings = timings;
+    pub fn timings(mut self, timings: &'a mut [LatencyTimingsFrameReportNV<'a>]) -> Self {
```

Comment:
It seems all lifetime-fu has escaped me; I can no longer use the slice passed as `timings` here even after `self` is dropped. Notably this is the first and only mutable slice that contains objects with a lifetime parameter.

```rust
let mut info = vk::GetLatencyMarkerInfoNV::default();
let latency_timings = base
    .low_latency2_loader
    .get_latency_timings_len(base.swapchain, &mut info);
dbg!(latency_timings);
if latency_timings != 0 {
    let mut timings = vec![vk::LatencyTimingsFrameReportNV::default(); latency_timings];
    {
        // Creating a new object just for show
        let mut info = vk::GetLatencyMarkerInfoNV::default().timings(&mut timings);
        base.low_latency2_loader.get_latency_timings(
            base.swapchain,
            latency_timings,
            &mut info,
        );
        // `info` should be dropped here, removing the mutable borrow on `timings`?
    }
    dbg!(&timings); // Error: `timings` is still borrowed in `.timings(&mut timings);` above?
}
```

```
error[E0502]: cannot borrow `timings` as immutable because it is also borrowed as mutable
   --> examples/src/bin/texture.rs:800:22
    |
793 |                     let mut info = vk::GetLatencyMarkerInfoNV::default().timings(&mut timings);
    |                                                                                  ------------ mutable borrow occurs here
...
800 |                 dbg!(&timings);
    |                      ^^^^^^^^
    |                      |
    |                      immutable borrow occurs here
    |                      mutable borrow later used here
```

Reply:
I think by constructing an `&'a mut [LatencyTimingsFrameReportNV<'a>]`, you're saying that the slice is uniquely borrowed for (`&'a mut`) a lifetime that outlives it (`T<'a>`). Does it work better if you do `for<'b: 'a> &'a mut [LatencyTimingsFrameReportNV<'b>]`?

Reply:
@Ralith that is indeed the first thing I tried, making sense that `'b` is a lifetime that is bigger / lives longer than `'a` (because `LatencyTimingsFrameReportNV` and the things that it borrows can live longer than the slice it is eventually stored within).

Unfortunately the compiler is not having it:

```rust
pub fn timings<'b: 'a>(mut self, timings: &'a mut [LatencyTimingsFrameReportNV<'b>]) -> Self {
    self.p_timings = if timings.is_empty() {
        std::ptr::null_mut()
    } else {
        timings.as_mut_ptr()
    };
    self
}
```

Results in:

```
error: lifetime may not live long enough
     --> ash/src/vk/definitions.rs:51910:13
      |
51892 | impl<'a> GetLatencyMarkerInfoNV<'a> {
      |      -- lifetime `'a` defined here
...
51906 |     pub fn timings<'b: 'a>(mut self, timings: &'a mut [LatencyTimingsFrameReportNV<'b>]) -> Self {
      |                    -- lifetime `'b` defined here
...
51910 |             timings.as_mut_ptr()
      |             ^^^^^^^^^^^^^^^^^^^^ argument requires that `'a` must outlive `'b`
      |
      = help: consider adding the following bound: `'a: 'b`
      = note: requirement occurs because of a mutable reference to `[vk::definitions::LatencyTimingsFrameReportNV<'_>]`
      = note: mutable references are invariant over their type parameter
      = help: see <https://doc.rust-lang.org/nomicon/subtyping.html> for more information about variance
```

Writing the bound the other way around results in the same error as before. And it doesn't make much sense for `'a` to outlive `'b` when `'a` is referring to a thing of shorter 
ifetime `'b`?